### PR TITLE
Add inspection strategies

### PR DESCRIPTION
**Pull request by kellynm**
Added percent stems strategy. Next step will be changing function structure.

### CODE REVIEW

**Comment by kellynm** [comment]
Inspect function outputs:
- % shipments slipped (false negative)
- Avg infestation rate of intercepted shipments, histogram of infestation rates
- Avg infestation rate of slipped shipments, histogram of infestation rates
- Avg # boxes opened, to detection and completion
- % boxes opened, to detection and completion
- Avg # stems inspection, to detection and completion
- % stems inspected, to detection and completion
- Avg % of sample completed if end inspection at detection
- Avg % of intercepted pest unreported if end inspection at detection

File: pathways/inspections.py [hunk 1]
```diff
@@ -68,65 +67,212 @@ def inspect_first_n(num_boxes, shipment):
...
-        return True, boxes_to_inspect
+    unit = config["inspection"]["unit"]
+    detection_level = config["inspection"]["hypergeometric"]["detection_level"]
+    confidence_level = config["inspection"]["hypergeometric"]["confidence_level"]
+    num_stems = shipment["num_stems"]
+    num_boxes = shipment["num_boxes"]
+
+    if unit =="stems":
+        n_stems_to_inspect = math.ceil((1-((1-confidence_level)**
+        (1/(detection_level*num_stems))))*(num_stems-(((detection_level*num_stems)-1)/2)))
```
Comment (wenzeslaus) [inline pathways/inspections.py:113]: This looks like a candidate for a function. It is complicated and it is there twice. If you have a(ny) name for it, just go ahead and make it a function.

File: pathways/inspections.py [hunk 2]
```diff
@@ -191,10 +337,16 @@ def shipment_infestation_rate(shipment):
...
     for box in shipment["boxes"]:
         if box:
             count += 1
     return count
+
+
+def count_diseased_stems(shipment):
+    """Return number of stems with pest"""
+    count = np.count_nonzero(shipment["stems"])
+    return count
```
Comment (wenzeslaus) [inline pathways/inspections.py:352]: File should end with a new line character both in Git and in Python following PEP8. See if your editor supports something like PEP8, pep8, pycodestyle, pyflakes, ... to catch these automatically. The code in this repo is checked [using Flake8, Black and Pylint](https://github.com/ncsu-landscape-dynamics/pathways-simulation/blob/master/.github/workflows/ci.yml) with configuration [in the root directory](https://github.com/ncsu-landscape-dynamics/pathways-simulation). However, don't worry about it too much right now. I think I will just merge this PR once we are happy with the code and we can work on fixing it later (Black will fix the whitespace formalities automatically).

File: pathways/simulation.py [hunk 3]
```diff
@@ -165,17 +218,41 @@ def run_simulation(
...
+        totals.pct_boxes_opened_detection=0,
+        totals.avg_stems_inspected_completion=0,
+        totals.avg_stems_inspected_detection=0,
+        totals.pct_stems_inspected_completion=0,
+        totals.pct_stems_inspected_detection=0,
+        totals.pct_sample_if_to_detection=0,
+        totals.pct_pest_unreported_if_detection=0,
+        totals.true_infestation_rate=0,
+        totals.missed_infestation_rate=0,
+        totals.intercepted_infestation_rate=0,
```
Comment (wenzeslaus) [inline pathways/simulation.py:255]: Unlike the other places, here should be no commas at the end of the line.

File: pathways/simulation.py [hunk 4]
```diff
@@ -189,14 +266,38 @@ def run_simulation(
...
+        totals.pct_boxes_opened_completion+=result.pct_boxes_opened_completion,
+        totals.pct_boxes_opened_detection+=result.pct_boxes_opened_detection,
+        totals.avg_stems_inspected_completion+=result.avg_stems_inspected_completion,
+        totals.avg_stems_inspected_detection+=result.avg_stems_inspected_detection,
+        totals.pct_stems_inspected_completion+=result.pct_stems_inspected_completion,
+        totals.pct_stems_inspected_detection+=result.pct_stems_inspected_detection,
+        totals.pct_sample_if_to_detection+=result.pct_sample_if_to_detection,
+        totals.pct_pest_unreported_if_detection+=result.pct_pest_unreported_if_detection,
+        totals.true_infestation_rate+=result.true_infestation_rate,
+        totals.missed_infestation_rate+=result.missed_infestation_rate,
```
Comment (wenzeslaus) [inline pathways/simulation.py:281]: Extra commas.

File: pathways/inspections.py (hunk 1 above)
Comment (kellynm) [reply pathways/inspections.py]: Added function.

File: pathways/simulation.py (hunk 3 above)
Comment (kellynm) [reply pathways/simulation.py]: Removed commas.

File: pathways/simulation.py (hunk 4 above)
Comment (kellynm) [reply pathways/simulation.py]: Also removed commas here.

File: pathways/inspections.py (hunk 2 above)
Comment (kellynm) [reply pathways/inspections.py]: Yes, the IDE I'm using has Pylint. Just went through with the linter and caught a bunch of bugs... Very helpful! There are still some issues (hopefully minor) I was not sure how to fix.

**Comment by wenzeslaus** [comment]
Merged with failing checks. I will resolve what I can and let's open issues for any further problems if needed.

**Pull request merged**